gateleen-hook/src/test/java/org/swisspush/gateleen/hook/HookHandlerTest.java

```diff
@@ -641,6 +643,333 @@ public void hookRegistration_RouteWithRouteMultiplier(TestContext testContext) t
+                            context.assertTrue(jsonResponse.getJsonArray("listeners").contains("x99"));
+                            context.assertFalse(jsonResponse.getJsonArray("listeners").contains("x100"));
+                            context.async().complete();
+                        })
+                        .onFailure(context::fail);
+            } else {
+                context.fail(ar.cause());
+            }
+        });
+    }
+    @Test
+    public void testSearchSingleListener_Success(TestContext context) {
+        Vertx vertx = Vertx.vertx();
+        MockResourceStorage storage = new MockResourceStorage();
+        HttpClient httpClient = vertx.createHttpClient();
+
+        // Create mock implementation for RequestQueue
+        RequestQueue requestQueue = Mockito.mock(RequestQueue.class);
+
+        // Initialize HookHandler
+        HookHandler hookHandler = new HookHandler(vertx, httpClient, storage, loggingResourceManager,
+                logAppenderRepository, monitoringHandler,
+                "userProfilePath", "/hookRootUri/",
+                requestQueue, false, reducedPropagationManager);
+
+        // Insert a single listener to the storage
+        JsonObject listener1 = new JsonObject()
+                .put("requesturl", "/playground/server/tests/hooktest/_hooks/listeners/http/push/listener1")
+                .put("expirationTime", "2025-01-03T14:15:53.277")
+                .put("hook", new JsonObject().put("destination", "/playground/server/push/v1/devices/listener1"));
+        storage.putMockData("/_hooks/listeners/listener1", listener1.encode());
+
+        // Configure HttpServer for integration test
+        io.vertx.ext.web.Router router = io.vertx.ext.web.Router.router(vertx);
+        router.route().handler(hookHandler::handle);
+
+        HttpServer server = vertx.createHttpServer();
+        server.requestHandler(router).listen(8080, ar -> {
+            if (ar.succeeded()) {
+                // Make a real HTTP request to the HookHandler
+                httpClient.request(HttpMethod.GET, 8080, "localhost", "/_hooks/listeners?q=listener1")
+                        .compose(HttpClientRequest::send)
+                        .compose(response -> {
+                            context.assertEquals(200, response.statusCode());
+                            return response.body();
+                        })
+                        .onSuccess(body -> {
+                            JsonObject jsonResponse = new JsonObject(body.toString());
+                            context.assertTrue(jsonResponse.getJsonArray("listeners").contains("listener1"));
+                            context.async().complete();
+                        })
+                        .onFailure(context::fail);
+            } else {
+                context.fail(ar.cause());
+            }
+        });
+    }
+    @Test
+    public void testSearchListenerNotFound_MultipleListeners_Failure(TestContext context) {
```

Comment:
Migrate/remove this test to `gateleen-test` module since we don't do integration tests here

Reply:
I moved the test to the tests module